src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs

```diff
@@ -0,0 +1,118 @@
+using System.IO;
+using GitVersion;
+using GitVersion.BuildAgents;
+using GitVersion.Core.Tests.Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+using Shouldly;
+
+namespace GitVersionCore.Tests.BuildAgents
+{
+    [TestFixture]
+    public class SpaceAutomationTests : TestBase
+    {
+        private IEnvironment environment;
+        private SpaceAutomation buildServer;
+        private string setEnvironmentTempFilePath;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var sp = ConfigureServices(services =>
+            {
+                services.AddSingleton<SpaceAutomation>();
+            });
+            environment = sp.GetService<IEnvironment>();
+            buildServer = sp.GetService<SpaceAutomation>();
+            environment.SetEnvironmentVariable(SpaceAutomation.EnvironmentVariableName, "true");
+
+            setEnvironmentTempFilePath = Path.GetTempFileName();
```

Comment:
I think this one is not needed here at all, as only the GH actions build agent is using a file, so it is needed only in the GH tests